Fix #4308: Fix all Session Restore, Reader-Mode, GCDWebServer issues

Brandon-T opened the pull request:
## Info:
- Fix all session restore, reader-mode, error page, certificate error page, etc. bugs.
- Used InternalURL and fix PrivilegedRequest.
- Get rid of most of GCDWebServer code.

<!-- *Thank you for submitting a pull request, your contributions are greatly appreciated!* -->

## Summary of Changes
- Use `InternalURL`
- Use `PrivilegedRequest` which uses `InternalURL`
- Remove most `GCDWebServer` handlers to use `WKURLSchemeHandler` instead
- Update ErrorPage HTML and CSS
- Update NetCert HTML and CSS
- Update SessionRestore HTML
- Rewrote all Interstitial pages
- Update SessionData and SavedTab to handle persistent private mode (though, not enabled yet)
- Added migration `SessionData.updateSessionURLs` to migrate session restore URLs to use the proper schemes instead of (`GCDWebServer` schemes && `localhost`).

- **TODO**: Update Unit Tests!! (Help wanted :))

<!-- Enter a ticket number for this PR, create a new one if it is not there yet. -->
This pull request fixes #4308

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Test Plan:
- Test Reader-Mode
- Test Session Restore
- Test Session Restore of Japanese website with Kanji or Katakana characters in the URL
- Test Session Restore of multiple tabs
- Test About Home page
- Test About License page
- Test Error Pages like on `gsdgsdglkangajkngkalngakgdaga.com` or random website.
- Test Error Pages on bad certificate website like: `https://expired.badssl.com/`
- Test Error Pages when there is no network (turn on airplane mode).


## Screenshots:
<!-- If your patch includes user interface changes that you would like to suggest or that you would like UX to look at, please include them here. -->


## Reviewer Checklist:

- [x] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [x] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [x] Adequate unit test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable).
- [x] Issue and pull request is assigned to a milestone (should happen at merge time).

Brandon-T committed:
Added Interstitial Pages

Update to show alternative Certificate Error page if there is no certificate (OR certificate uses invalid cipher or algorithms).

iccub reviewed (changes requested):
left few questions and nitpicks, will run this branch later for functional review

Brandon-T committed:
Fixing Unit Tests.

Brandon-T committed:
Addressing feedback

Brandon-T committed:
Update `isLocalUtility`

Brandon-T committed:
Removing code we got from Firefox for Progress Bar update on tab switch.

iccub reviewed (approved):
I ran some tests and all works for me :)

Brandon-T committed:
Fix all session restore, reader-mode, error page, certificate error page, etc. bugs.
Used InternalURL and fix PrivilegedRequest.
Get rid of most of GCDWebServer code.

Brandon-T committed:
Fix unit test
